bindings/python/examples/iio_writedev.py

```diff
@@ -0,0 +1,221 @@
+        sys.stderr.write(str(err))
+        sys.exit(1)
+
+    arg_ip = ''
+    arg_uri = ''
+    scan_for_context = False
+    buffer_size = 256
+    num_samples = 0
+    timeout = 0
+    cyclic = False
+
+    for argument, value in arguments:
+        if argument in ('-h', '--help'):
+            usage()
+            exit(0)
+        elif argument in ('-n', '--network'):
+            arg_ip = str(value)
+        elif argument in ('-u', '--uri'):
+            arg_uri = str(value)
+        elif argument in ('-a', '--auto'):
+            scan_for_context = True
+        elif argument in ('-b', '--buffer-size'):
+            buffer_size = int(value)
+        elif argument in ('-s', '--samples'):
+            num_samples = int(value)
+        elif argument in ('-T', '--timeout'):
+            timeout = int(timeout)
+        elif argument in ('-c', '--cyclic'):
+            cyclic = True
+
+    device_name = get_device_and_channels(raw_arguments)[0]
+    channels = get_device_and_channels(raw_arguments)[1:]
+
+    if device_name is None:
+        sys.stderr.write('Incorrect number of arguments.')
+        usage()
+        exit(1)
+
+    ctx = None
+
+    try:
+        if scan_for_context:
+            contexts = iio.scan_contexts()
+            if len(contexts) == 0:
+                sys.stderr.write('No IIO context found.')
+                exit(1)
+            elif len(contexts) == 1:
+                uri, descriptor = contexts.popitem()
+                ctx = iio.Context(_context=uri)
+            else:
+                print('Multiple contexts found. Please select one using --uri!')
+
+                for uri, descriptor in contexts:
+                    print(uri)
+        elif arg_uri != '':
+            ctx = iio.Context(_context=arg_uri)
+        elif arg_ip != '':
+            ctx = iio.NetworkContext(arg_ip)
+        else:
+            ctx = iio.Context()
+    except FileNotFoundError:
+        sys.stderr.write('Unable to create IIO context')
+        exit(1)
+
+    if timeout >= 0:
+        ctx.set_timeout(timeout)
+
+    dev = ctx.find_device(device_name)
+
+    if dev is None:
+        sys.stderr.write('Device %s not found!' % device_name)
+        exit(1)
+
+    if len(channels) == 0:
+        for channel in dev.channels:
+            channel.enabled = True
+    else:
+        for channel_idx in channels:
+            dev.channels[int(channel_idx)].enabled = True
+
+    buffer = iio.Buffer(dev, buffer_size, cyclic=cyclic)
+
+    if buffer is None:
+        sys.stderr.write('Unable to create buffer!')
+        exit(1)
+
+    app_running = True
+    num_samples_set = num_samples > 0
+
+    while app_running:
+        bytes_to_read = buffer._length if not num_samples_set else min(buffer._length, num_samples * dev.sample_size)
+        write_len = bytes_to_read
+        data = []
+
+        while bytes_to_read > 0:
+            data = sys.stdin.buffer.read(bytes_to_read)
+            bytes_to_read -= len(data)
```

Comment:
Won't this loop simply overwrite data when bytes_to_read is not 0 after the first loop? Therefore the first array is never passed to the write buffer.

Reply:
Yes, you are right.  I have modified; I think this modification should solve the problem.